airbyte-cdk/python/unit_tests/sources/test_abstract_source.py

```diff
@@ -1297,6 +1303,71 @@ def test_emit_non_records(self, mocker):
 
         assert messages == expected
 
+    @pytest.mark.parametrize(
+        "use_legacy",
```

Comment:
I would actually say we can drop this `use_legacy` scenario in this test. Legacy state is rarely used anymore and it doesn't impact the root scenario of what the new test is trying to validate anyway

Reply:
Aaaaaand is gone.